src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,146 @@
+package duke.command;
+
+import duke.component.DukeDatabase;
+import duke.component.TaskList;
+import duke.component.Ui;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Event;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Encapsulates a command that adds a task into the tasks list of Duke.
+ */
+public class AddCommand extends Command {
+    private AddType addType;
+
+    /**
+     * Represents the type of add command.
+     * TODO: to add a ToDo Task, DEADLINE: to add a Deadline Task, EVENT: to add an Event task.
+     */
+    public enum AddType {
+        TODO, DEADLINE, EVENT;
+    }
+
+    /**
+     * Constructs an AddCommand object.
+     *
+     * @param addType the type of adding command.
+     * @param input the user's input.
+     */
+    public AddCommand(AddType addType, String input) {
+        super(input);
+        this.addType = addType;
+    }
+
+    /**
+     * Executes the adding command accordingly.
+     *
+     * @param tasksList the tasks list of Duke.
+     * @param ui the ui of Duke.
+     * @param database the database of Duke.
+     * @throws DukeException if the user's input is incorrect.
+     */
+    public void execute(TaskList tasksList, Ui ui, DukeDatabase database) throws DukeException {
+        initialise(tasksList, ui, database);
+
+        if (AddType.TODO.equals(addType)) {
+            addToDo();
+        } else if (AddType.DEADLINE.equals(addType)) {
+            addDeadline();
+        } else if (AddType.EVENT.equals(addType)) {
+            addEvent();
+        } else {
+            throw new DukeException("Internal logic bug occurred!");
+        }
+    }
+
+    /**
+     * Adds to do entry to the taskList.
+     */
+    private void addToDo() throws DukeException {
+        String topic = input.substring(4).trim();
```

Comment:
An alternative to String manipulation would be to use Regular Expressions. You can refer to my fork in Duke.java to see how it could be used. As the complexity of the commands increase, writing equivalent String Manipulation code would become increasingly tedious.

In our use case, the commands are trivial, and would likely be faster than using Regex, however the tradeoff would be a little more refactoring in the future when commands become more complex.